clang/lib/Basic/OffloadArch.cpp

```diff
@@ -0,0 +1,128 @@
+#include "clang/Basic/OffloadArch.h"
+
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/StringRef.h"
+
+namespace clang {
+
+namespace {
+struct OffloadArchToStringMap {
+  OffloadArch Arch;
+  const char *ArchName;
+  const char *VirtualArchName;
+};
+} // namespace
+
+#define SM2(sm, ca) {OffloadArch::SM_##sm, "sm_" #sm, ca}
```

Comment:
I'd just define the special `sm_20` w/o using this macro

Reply:
Good point, that is simpler - [bd5e7a2](https://github.com/llvm/llvm-project/pull/137070/commits/bd5e7a26f48a1afcdfdb4e14415572a0d2a72875)